File: Behavioral/Iterator/Python/song_collection.py
# song_collection.py

from song_iterator import SongIterator

class SongCollection:
  def __init__(self):
    self._songs = []

  def add_song(self, song):
    self._songs.append(song)

  def get_iterator(self):
    return SongIterator(self._songs)

File: Behavioral/Iterator/CSharp/SongCollection.cs
// SongCollection.cs

using System.Collections.Generic;

public class SongCollection {
  private readonly List<Song> _songs = new List<Song>();

  public void AddSong(Song song) {
    _songs.Add(song);
  }

  public IIterator<Song> GetIterator() {
    return new SongIterator(_songs);
  }
}
